src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1932,6 +1932,8 @@ public JPopupMenu getComponentPopupMenu() {
         if (viewMenu != null) {
             viewMenu.getPopupMenu().setInvoker(viewMenu);
         }
+
+        contextMenu.applyComponentOrientation(getFileChooser().getComponentOrientation());
```

Comment:
I see that this bug is old. Seems like a simple way to fix this and I see similar calls for other UI elements, like JMenu. It looks like in JMenu, it iterates through all components and sets it orientation, but it looks like the popup is the only problem, so looks fine.

You still need to update the year in the header of FilePane.

Reply:
Yeah, `applyComponentOrientation ` iterates through all the child components and sets the orientation.